Unable to build iOS app with extension .NET9

### Apple platform

iOS

### Framework version

net9.0-*

### Affected platform version

Visual Studio 17.14.18

### Description

After migrating a .NET MAUI solution from .NET 8 to .NET 9, I am unable to build the iOS project when it includes an app extension (e.g. Share Extension / Widget Extension).

- If I remove the app extension project reference from the main MAUI project, the build succeeds.
- The app extension project also builds successfully on its own.
- The failure only occurs when building the main MAUI iOS project with the app extension included as a Project Reference.

The build appears to fail during the _CollectCodesigningData MSBuild task:

```
    Target _CollectCodesigningData:
      Task "ReadItemsFromFile"
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018: The "ReadItemsFromFile" task failed unexpectedly.
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018: System.Xml.XmlException: Root element is missing.
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.Throw(Exception e)
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at System.Xml.XmlTextReaderImpl.ParseDocumentContent()
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at System.Xml.Linq.XDocument.Load(XmlReader reader, LoadOptions options)
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at System.Xml.Linq.XDocument.Load(Stream stream, LoadOptions options)
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at Xamarin.MacDev.Tasks.ReadItemsFromFile.Execute() in /Users/builder/azdo/_work/1/s/macios/msbuild/Xamarin.MacDev.Tasks/Tasks/ReadItemsFromFile.cs:line 45
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Execute()
        C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk.net9.0_26.0\26.0.9752\tools\msbuild\Xamarin.Shared.targets(2367,3): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext()
      Done executing task "ReadItemsFromFile" -- FAILED.
```

### Steps to Reproduce

Build a MAUI solution with an iOS app extension referenced in the main MAUI project with valid code-signing and entitlements on both.

### Did you find any workaround?

_No response_

### Build logs

_No response_

--- a/msbuild/Xamarin.Shared/Xamarin.Shared.targets
+++ b/msbuild/Xamarin.Shared/Xamarin.Shared.targets
@@ -405,8 +405,8 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 			<_IpaPackageFile Include="$(DeviceSpecificOutputPath)*.ipa" />
 		</ItemGroup>
 
-		<Delete SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Files="$(DeviceSpecificOutputPath)codesign.items" />
-		<Delete SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Files="$(DeviceSpecificOutputPath)codesign-bundle.items" />
+		<Delete Condition="'$(IsMacEnabled)' == 'true'" Files="$(DeviceSpecificOutputPath)codesign.items" />
+		<Delete Condition="'$(IsMacEnabled)' == 'true'" Files="$(DeviceSpecificOutputPath)codesign-bundle.items" />
 		<Delete SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Files="$(DeviceSpecificOutputPath)native-frameworks.items" />
 		<Delete SessionId="$(BuildSessionId)" Condition="'$(IsMacEnabled)' == 'true'" Files="@(_IpaPackageFile)" />
 	</Target>
@@ -2531,7 +2531,6 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 
 		<WriteItemsToFile
 			Condition="'$(IsMacEnabled)' == 'true'"
-			SessionId="$(BuildSessionId)"
 			Items="@(_CodesignBundle)"
 			ItemName="_CodesignBundle"
 			File="$(DeviceSpecificOutputPath)codesign-bundle.items"
@@ -2541,7 +2540,6 @@ Copyright (C) 2018 Microsoft. All rights reserved.
 
 		<WriteItemsToFile
 			Condition="'$(IsMacEnabled)' == 'true'"
-			SessionId="$(BuildSessionId)"
 			Items="@(_CodesignItems)"
 			ItemName="_CodesignItems"
 			File="$(DeviceSpecificOutputPath)codesign.items"

